src/Console/Helper/DumpInfo.php

```diff
@@ -0,0 +1,181 @@
+     */
+    private function getDumpFinishedListener(): callable
+    {
+        return function (): void {
+            if ($this->lastTableInfo) {
+                // Display information of the last table that was dumped
+                $this->updateProgressBarMessage($this->lastTableInfo);
+            }
+
+            $this->progressBar->finish();
+            $this->output->writeln('');
+        };
+    }
+
+    /**
+     * Get the hook that will be triggered when an object is being dumped.
+     */
+    private function getDumpInfoHook(): callable
+    {
+        return function (string $object, array $info): void {
+            if ($object !== 'table') {
+                // The max steps of the progress bar only include tables
+                return;
+            }
+
+            if ($info['completed']) {
+                $this->lastTableInfo = $info;
+                return;
+            }
+
+            $this->updateProgressBarMessage($info);
+
+            $info['rowCount'] === 0
+                ? $this->progressBar->advance() // new table
+                : $this->progressBar->setProgress($this->progressBar->getProgress()); // refresh current table progress
+        };
+    }
+
+    /**
+     * Display a section.
+     */
+    private function displaySection(string $name): void
+    {
+        $this->output->writeln(sprintf('<comment>%s</comment>', $name));
+    }
+
+    /**
+     * Display a section item.
+     */
+    private function displaySectionItem(string $name, mixed $value): void
+    {
+        $this->output->writeln(sprintf('  - %s: <info>%s</info>', $name, $this->formatValue($value)));
+    }
+
+    /**
+     * Update the progress bar message.
+     */
+    private function updateProgressBarMessage(array $tableInfo): void
+    {
+        $message = sprintf('<info>%s</info> (%s)', $tableInfo['name'], $this->formatRowCount($tableInfo['rowCount']));
+        $this->progressBar->setMessage($message, 'title');
+    }
+
+    /**
+     * Get max number of steps of the progress bar.
+     */
+    private function getMaxSteps(DumperConfig $config, MetadataInterface $metadata): int
+    {
+        $includedTables = $config->getTablesWhitelist() ?: $metadata->getTableNames();
+        $excludedTables = $config->getTablesBlacklist();
+
+        return count(array_diff($includedTables, $excludedTables));
+    }
+
+    /**
+     * Format a value for console output.
+     */
+    private function formatValue(mixed $value): string
+    {
+        if (is_bool($value)) {
+            return $value ? 'true' : 'false';
+        }
+
+        if (is_scalar($value)) {
+            return (string) $value;
+        }
+
+        return json_encode($value, JSON_THROW_ON_ERROR | JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
+    }
+
+    /**
+     * Format the row count.
+     */
+    private function formatRowCount(int $rowCount): string
+    {
+        return $rowCount > 1 ? sprintf('%d rows', $rowCount) : sprintf('%d row', $rowCount);
```

Comment:
what do you think about a thousands separator?

Reply:
With a space character as the separator of course :smiley: